common/keyboard.go

```diff
@@ -242,6 +292,74 @@ func (k *Keyboard) modifierBitFromKeyName(key string) int64 {
 	return 0
 }
 
+func (k *Keyboard) comboPress(keys string, opts *KeyboardOptions) error {
+	if opts.Delay != 0 {
+		t := time.NewTimer(time.Duration(opts.Delay) * time.Millisecond)
+		select {
+		case <-k.ctx.Done():
+			if !t.Stop() {
+				<-t.C
+			}
+			return context.Canceled
+		case <-t.C:
+		}
+	}
+
+	kk := split(keys)
+	for _, key := range kk {
+		if err := k.down(key); err != nil {
+			return fmt.Errorf("cannot do key down: %w", err)
+		}
+	}
+
+	for i := range kk {
+		key := kk[len(kk)-i-1]
+		if err := k.up(key); err != nil {
+			return fmt.Errorf("cannot do key up: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// This splits the string on `+`.
+// If `+` on it's own is passed, it will return ["+"].
+// If `++` is passed in, it will return ["+", ""].
+// If `+++` is passed in, it will return ["+", "+"].
+func split(keys string) []string {
+	kk := []string{}
+	var ss string
+	justAdded := true
+	for _, k := range keys {
+		var sb strings.Builder
+		sb.WriteRune(k)
+		s := sb.String()
+
+		switch {
+		case s == "+" && ss != "":
+			kk = append(kk, ss)
+			ss = ""
+			justAdded = true
+		case s == "+":
+			if justAdded {
+				kk = append(kk, "+")
+				ss = ""
+				justAdded = false
+			} else {
+				justAdded = true
+			}
+		default:
+			ss += s
+			justAdded = false
+		}
+	}
+	if ss != "" || justAdded {
+		kk = append(kk, ss)
+	}
+
+	return kk
+}
+
```

Comment:
Isn't it more like this?

```go
func split(keys string) []string {
	var (
		kk = make([]string, 0, len(keys))
		s  strings.Builder
	)
	for _, r := range keys {
		if r == '+' && s.Len() > 0 {
			kk = append(kk, s.String())
			s.Reset()
		} else {
			s.WriteRune(r)
		}
	}
	kk = append(kk, s.String())

	return kk
}
```

See: https://go.dev/play/p/fJoqZIMTBF5

Reply:
Nice! This is much easier to understand. Can you link me to the PW code please?